crates/telio-traversal/src/batcher.rs

```diff
@@ -490,153 +530,278 @@ mod tests {
+                Duration::from_millis((total_diff / count as i64) as u64)
             }
 
-            fn average_difference(numbers: &[i64]) -> f64 {
+            fn average_difference(numbers: &[Duration]) -> Duration {
                 if numbers.len() < 2 {
                     panic!("not enough elements");
                 }
 
-                let total_diff: i64 = numbers.windows(2).map(|w| w[0] - w[1]).sum();
+                let total_diff: i64 = numbers
+                    .windows(2)
+                    .map(|w| (w[0].as_millis() as i64 - w[1].as_millis() as i64).abs())
+                    .sum();
 
                 let count = (numbers.len() - 1) as f64;
-                total_diff as f64 / count
+                Duration::from_millis((total_diff / count as i64) as u64)
             }
 
-            let (first_half_a, second_half_a) =
-                left_values.as_slice().split_at(left_values.len() / 2);
-            let (first_half_b, second_half_b) =
-                right_values.as_slice().split_at(right_values.len() / 2);
-
-            TestHelperResult {
-                left_peer_avg_interval: average_difference(left_values.as_slice()),
-                right_peer_avg_interval: average_difference(right_values.as_slice()),
+            let res_1st = TestHelperResult {
+                left_peer_avg_interval: average_difference(left_1st_values.as_slice()),
+                right_peer_avg_interval: average_difference(right_1st_values.as_slice()),
 
-                left_peer_avg_interval_1st_half: average_difference(first_half_a),
-                left_peer_avg_interval_2nd_half: average_difference(second_half_a),
+                avg_aligned_interval: average_alignment(
+                    left_1st_values.as_slice(),
+                    right_1st_values.as_slice(),
+                ),
+            };
 
-                right_peer_avg_interval_1st_half: average_difference(first_half_b),
-                right_peer_avg_interval_2nd_half: average_difference(second_half_b),
+            let res_2nd = TestHelperResult {
+                left_peer_avg_interval: average_difference(left_2nd_values.as_slice()),
+                right_peer_avg_interval: average_difference(right_2nd_values.as_slice()),
 
                 avg_aligned_interval: average_alignment(
-                    left_values.as_slice(),
-                    right_values.as_slice(),
+                    left_2nd_values.as_slice(),
+                    right_2nd_values.as_slice(),
                 ),
-                avg_aligned_interval_1st_half: average_alignment(first_half_a, first_half_b),
-                avg_aligned_interval_2nd_half: average_alignment(second_half_a, second_half_b),
-            }
+            };
+
+            return (res_1st, res_2nd);
+        }
+
+        fn durations_close_enough(d1: Duration, d2: Duration, tolerance: Duration) -> bool {
+            let diff = if d1 > d2 { d1 - d2 } else { d2 - d1 };
```

Comment:
`abs_diff` should work here too

Reply:
```
error[E0658]: use of unstable library feature 'duration_abs_diff'
   --> crates/telio-traversal/src/batcher.rs:623:16
    |
623 |             d1.abs_diff(d2) <= tolerance
    |                ^^^^^^^^
    |
```

Leaving it as it is because of this